src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,80 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int WORKED_HOURS_INDEX = 2;
+    private static final int RATE_INDEX = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate from;
+        LocalDate to;
+        try {
+            from = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+            to = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+        } catch (DateTimeParseException e) {
+            throw new RuntimeException("Can not parse entry date", e);
+        }
+        List<String> dataList = new ArrayList<>(new ArrayList<>(Arrays.asList(data)));
+        int[] countedSalary = countSalary(names, from, to, dataList);
+        StringBuilder stringBuilder = new StringBuilder();
+        appendReportHeader(dateFrom, dateTo, stringBuilder);
+        return appendSalaryInfo(names, stringBuilder, countedSalary);
+    }
```

Comment:
```suggestion
        String dataFromFile = readFromFile(fromFileName);
        String report = createReport(dataFromFile);
        writeToFile(report, toFileName)
    }
```
lets divide the logic into 3 methods - to make your code easier to read

Reply:
ОK